### ISSUE TITLE
Itinerary covering nearby places

### ISSUE THREAD

**Issue by nicolas-raoul** [commons-app/apps-android-commons#1118]
I stumbled upon this old app that does not work anymore since 2015:
https://commons.wikimedia.org/wiki/Commons:Unvisited_app

I find it interesting that they suggest an itinerary to cover all points of interest in the area:

![unvisited_trip_map_around_elm_street_historic_district _new_haven_ct](https://user-images.githubusercontent.com/99590/35669725-65c6ff00-0779-11e8-85bb-8c8e0058ae55.png)

Some ideas:
- Ask the user how much time they have and how fast they can move. Example: 10 minutes, 30 minutes, 1 hour, 3 hours, 6 hours, 9 hours / Walking, Cycling, Driving. Use that information to cover as many points as possible
- Rather than a simple https://en.wikipedia.org/wiki/Travelling_salesman_problem use real itinerary planning using a service that allows it (maybe https://maps.openrouteservice.org https://github.com/GIScience/graphhopper https://github.com/graphhopper/jsprit ?). I asked at https://discuss.graphhopper.com/t/visit-as-many-points-as-possible-within-certain-time/2778

**Comment by neslihanturan** [comment]
I think displayin all points (according to our current algorithm), but using a different color for closer (according to time/distance constraints) ones would be nice. If we only display reachable points,  we may miss some points that are a little far from users distance estimation. It can be hard for user define the constraints sometimes. 

Also we can consider this feature as a part of our gamification discussion.

**Comment by nicolas-raoul** [comment]
@misaochan Yes our license does not prevent us from calling any API we want, even proprietary.

Getting familiar with the API of graphhopper and the related services would take some time, so better leave the task to any volunteer who is very interested in GIS.

**Comment by misaochan** [comment]
@maskaravivek Great idea! Needs to have #591 done first but after that this would be very useful IMO.

**Comment by misaochan** [comment]
Btw, @maskaravivek could you create a new issue for your suggestion please? :) it is quite different from Nicolas 's.

**Comment by maskaravivek** [comment]
@misaochan As suggested, I have created a new issue for it [#1499].

Sorry, I could not understand why #591 needs to be done before implementing this. I was thinking of an offline list that would not be dependent on the APIs. Moreover, the list could contain all items bookmarked by him sorted by the distance.

**Comment by nicolas-raoul** [comment]
I forgot I had created this issue, and ended up writing a more detailed one at https://github.com/commons-app/apps-android-commons/issues/5671, so closing this one.
